docs/mypy.md

```diff
@@ -0,0 +1,30 @@
+# mypy static checker
+
+Zulip now uses mypy. mypy is a compile-time type checker for python.
+For information about what mypy is, visit [http://mypy-lang.org/](http://mypy-lang.org/).
+
+mypy's type checking has been added to Zulip's Travis CI checks.
+
+## Running mypy on Zulip's code
+
+To run mypy on Zulip's code, you'll need to make sure that mypy is installed.
+mypy only runs on python 3. Zulip doesn't use the latest version of mypy available on pypi.
+It uses a specific commit from [mypy's git repository](https://github.com/python/mypy).
+
+*   If you installed Zulip's development environment using vagrant or provision.py,
+    mypy is already installed. To switch to an appropriate python 3 virtual environment, run
+
+        source /srv/zulip-py3-venv/bin/activate
+
+*   If you have not installed Zulip using the above methods, you will have to manually set up
+    a python 3 virtual environment and install mypy in it. Run the following code from Zulip's
+    root directory:
+
+        virtualenv -p python3 ../zulip-py3-venv
+        source ../zulip-py3-venv/bin/activate
+        pip install --upgrade pip
+        pip install -r py3_test_reqs.txt
```

Comment:
When we achieve python 3 compatibility, the by-hand instructions in `README.dev.md` will look like this:
```
# Since Zulip supports python 2 and python 3,
# install a separate virtualenv for each of them.
virtualenv -p python2 ../zvenv2
source ../zvenv2/bin/activate
pip install --upgrade pip
pip install --no-deps -r reqs2.txt
virtualenv -p python3 ../zvenv3
source ../zvenv3/bin/activate
pip install --upgrade pip
pip install --no-deps -r reqs3.txt
```

This is not much different from what I'm doing now. We might get away in the meantime but eventually we'll face this problem.

I agree that we should have an easier way to do this, but I'm yet to find that easier way.